pulsar-sql/presto-pulsar/src/main/java/org/apache/pulsar/sql/presto/PulsarConnectorCache.java

```diff
@@ -112,7 +112,7 @@ private static ManagedLedgerFactory initManagedLedgerFactory(PulsarConnectorConf
                 pulsarConnectorConfig.getManagedLedgerNumWorkerThreads());
         managedLedgerFactoryConfig.setNumManagedLedgerSchedulerThreads(
                 pulsarConnectorConfig.getManagedLedgerNumSchedulerThreads());
-        return new ManagedLedgerFactoryImpl(bkClientConfiguration, managedLedgerFactoryConfig);
+        return new ManagedLedgerFactoryImpl(bkClientConfiguration, pulsarConnectorConfig.getZookeeperUri(),managedLedgerFactoryConfig);
```

Comment:
> The code `.setZkServers(pulsarConnectorConfig.getZookeeperUri())` could be removed in line 97.

please don't remove it； it 's very essentia for us to get manage-ledger info(/managed-ledger/) or get cursor info   because it need   zk root path();  Only when we create bookkeeper, we don't need ZK root path；ZK root path  will be auto created in bookeeper by metadataServiceUri(project bookkeeper AbstractZkLedgerManager.java row 169)

Reply:
setZKServers is diffrent from setMetadataServiceUri,we need both

Reply:
The `ClientConfiguration` bkClientConfiguration is used to generate the bookkeeper, the `MetadataServiceUri` is enough, we don't need to set the `zkServers`.

Reply:
yes,you are right.;

Reply:
before i revise it,it is neccessary to create `metadatastore`(ManagedLedgerFacoryImpl.java row 199);

Reply:
> The `ClientConfiguration` bkClientConfiguration is used to generate the bookkeeper, the `MetadataServiceUri` is enough, we don't need to set the `ZkServers` and the method `setZkServers` is deprecated.

today,i found it's neccessay to setZkServers .when broker's zk Path is diffrent from bookies's zk Path.